src/main/java/frc/robot/subsystems/scoring/AimerIORoboRio.java

```diff
@@ -0,0 +1,179 @@
+
+public class AimerIORoboRio implements AimerIO {
+    private final TalonFX aimerLeft = new TalonFX(ScoringConstants.aimLeftMotorId);
+    private final TalonFX aimerRight = new TalonFX(ScoringConstants.aimRightMotorId);
+
+    private final PIDController controller =
+            new PIDController(
+                    ScoringConstants.aimerkP, ScoringConstants.aimerkI, ScoringConstants.aimerkD);
+    private ArmFeedforward feedforward =
+            new ArmFeedforward(
+                    ScoringConstants.aimerkS,
+                    ScoringConstants.aimerkG,
+                    ScoringConstants.aimerkV,
+                    ScoringConstants.aimerkA);
+    private TrapezoidProfile profile =
+            new TrapezoidProfile(
+                    new TrapezoidProfile.Constraints(
+                            ScoringConstants.aimCruiseVelocity, ScoringConstants.aimAcceleration));
+
+    private final DutyCycleEncoder encoder = new DutyCycleEncoder(ScoringConstants.aimEncoderPort);
+
+    private final Timer timer = new Timer();
+
+    private boolean override = false;
+    private double overrideVolts = 0.0;
+
+    boolean newProfile = false;
+    double previousGoalAngle = 0.0;
+
+    double minAngleClamp = 0.0;
+    double maxAngleClamp = 0.0;
+
+    double goalAngleRad = 0.0;
+    double appliedVolts = 0.0;
+
+    double initialAngle = 0.0;
+    double initialVelocity = 0.0;
+
+    double velocity = 0.0;
+
+    double lastPosition = 0.0;
+    double lastTime = Utils.getCurrentTimeSeconds();
+
+    public AimerIORoboRio() {
+        aimerLeft.setControl(new Follower(ScoringConstants.aimRightMotorId, true));
+
+        aimerLeft.setNeutralMode(NeutralModeValue.Brake);
+        aimerRight.setNeutralMode(NeutralModeValue.Brake);
+
+        TalonFXConfigurator aimerLeftConfig = aimerLeft.getConfigurator();
+        aimerLeftConfig.apply(
+                new CurrentLimitsConfigs()
+                        .withStatorCurrentLimit(120)
+                        .withStatorCurrentLimitEnable(true));
+
+        TalonFXConfigurator aimerRightConfig = aimerRight.getConfigurator();
+        aimerRightConfig.apply(
+                new CurrentLimitsConfigs()
+                        .withStatorCurrentLimit(120)
+                        .withStatorCurrentLimitEnable(true));
+
+        aimerRight.setPosition(0.0);
+    }
+
+    @Override
+    public void setAimAngleRad(double goalAngleRad, boolean newProfile) {
+        this.goalAngleRad = goalAngleRad;
+        this.newProfile = newProfile;
+    }
+
+    @Override
+    public void controlAimAngleRad() {
+        if (goalAngleRad != previousGoalAngle && newProfile) {
+            timer.reset();
+            timer.start();
+
+            initialAngle = getEncoderPosition();
+            initialVelocity = velocity;
+
+            previousGoalAngle = goalAngleRad;
+        }
+        goalAngleRad = MathUtil.clamp(goalAngleRad, minAngleClamp, maxAngleClamp);
+    }
```

Comment:
It seems like we should be able to do this as 1 function since the actual controller lives in updateInputs. Also we can likely remove newProfile as stated in the other comment